sdk/python/kfp/compiler/_op_to_template.py

```diff
@@ -250,8 +250,38 @@ def _op_to_template(op: BaseOp):
         if processed_op.pod_labels:
             template['metadata']['labels'] = processed_op.pod_labels
     # retries
-    if processed_op.num_retries:
-        template['retryStrategy'] = {'limit': processed_op.num_retries}
+    if processed_op._node_retry_strategy:
+        retry_strategy = op._node_retry_strategy
+        retry_strategy_dict = {}
+
+        if retry_strategy.num_retries:
+            retry_strategy_dict['limit'] = retry_strategy.num_retries
+
+        backoff_dict = {}
+        if retry_strategy.backoff_duration:
+            backoff_dict['duration'] = retry_strategy.backoff_duration
+        if retry_strategy.backoff_factor:
+            backoff_dict['factor'] = retry_strategy.backoff_factor
+        if retry_strategy.backoff_max_duration:
+            backoff_dict['maxDuration'] = retry_strategy.backoff_max_duration
+        if backoff_dict:
+            retry_strategy_dict['backoff'] = backoff_dict
+
+        if retry_strategy.retry_on_error:
+            if retry_strategy.retry_on_failure:
+                retry_policy = 'Both' # Argo's constant
```

Comment:
Can we just let user to specify the retry_policy here?

Reply:
+1

Reply:
Usually I prefer not to use magic strings. Especially just exposing the strings used by another product without an indirection layer.